compiler/src/lnntoamm/Expr.ts

```diff
@@ -356,16 +356,23 @@ export default abstract class Expr {
             precedences[applyIdx] = dec.ref();
           }
           const applyTo = precedences[applyIdx] as Ref;
-          const fns = operators.reduce(
-            (fns, op) => [...fns, ...op.select(metadata.scope, applyTo.ty)],
-            new Array<Fn>(),
+          const [fns, retTys] = operators.reduce(
+            ([fns, retTys], op) => {
+              const [selFns, selTys] = op.select(metadata.scope, applyTo.ty);
+              fns = [...fns, ...selFns];
+              retTys = [...retTys, ...selTys];
+              return [fns, retTys];
+            },
+            [new Array<Fn>(), new Array<Type>()] as [Fn[], Type[]],
```

Comment:
What a failure of TS's type inference here...

Reply:
er, it does work without the `as` but it wasn't in a previous commit. removing the `as`